website/source/docs/guides/agent-encryption.html.md

```diff
@@ -0,0 +1,205 @@
+
+## Gossip Encryption
+
+To enable gossip encryption, you need to use an encryption key when starting the Consul agent. The key can be set via the `encrypt` parameter in the agent configuration file or in a separate file containing the encryption key. The key must be 16-bytes, Base64 encoded. 
+
+You can use the Consul CLI command, [`consul keygen`](/docs/commands/keygen.html), to generate a cryptographically suitable key.
+
+```sh
+$ consul keygen
+cg8StVXbQJ0gPvMd9o7yrg==
+```
+
+### Enable Gossip Encryption: New Cluster
+
+To enable gossip on a new cluster, we will add the encryption key parameter to the
+agent configuration file and then pass the file at startup with the [`-config-dir`](https://www.consul.io/docs/agent/options.html#_config_dir) flag.
+
+```javascript
+{
+  "data_dir": "/opt/consul",
+  "log_level": "INFO",
+  "node_name": "bulldog",
+  "server": true,
+  "encrypt": "JY34uTPZyfUE+6tinMYEVw=="
+}
+```
+
+```sh
+$ consul agent -config-dir=/etc/consul.d/
+==> Starting Consul agent...
+==> Starting Consul agent RPC...
+==> Consul agent running!
+         Node name: 'Armons-MacBook-Air.local'
+        Datacenter: 'dc1'
+            Server: false (bootstrap: false)
+       Client Addr: 127.0.0.1 (HTTP: 8500, HTTPS: -1, DNS: 8600, RPC: 8400)
+      Cluster Addr: 10.1.10.12 (LAN: 8301, WAN: 8302)
+    Gossip encrypt: true, RPC-TLS: false, TLS-Incoming: false
+...
+```
+
+"Encrypt: true" will be included in the output, if encryption is properly configured.
+
+Note: all nodes within a cluster must share the same encryption key in order to send and receive cluster information, including clients and servers. Additionally, if you're using multiple WAN joined datacenters, be sure to use _the same encryption key_ in all datacenters.
+
+### Enable Gossip Encryption: Existing Cluster
+
+Gossip encryption can also be enabled on an existing cluster, but requires several extra steps. The additional configuration of the agent configuration parameters, [`encrypt_verify_incoming`](/docs/agent/options.html#encrypt_verify_incoming) and [`encrypt_verify_outgoing`](/docs/agent/options.html#encrypt_verify_outgoing) is necessary. 
+
+**Step 1**: Generate an encryption key using `consul keygen`.
+
+```sh
+$ consul keygen
+JY34uTPZyfUE+6tinMYEVw==
+```
+
+**Step 2**: Set the [`encrypt`](/docs/agent/options.html#_encrypt) key, and set `encrypt_verify_incoming` and `encrypt_verify_outgoing` to `false` in the agent configuration file. Then initiate a rolling update of the cluster with these new values. After this step, the agents will be able to decrypt gossip but will not yet be sending encrypted traffic.
+
+```javascript
+{
+  "data_dir": "/opt/consul",
+  "log_level": "INFO",
+  "node_name": "bulldog",
+  "server": true,
+  "encrypt": "JY34uTPZyfUE+6tinMYEVw==",
+  "encrypt_verify_incoming": false,
+  "encrypt_verify_outgoing": false
+}
+```
+
+A rolling update can be made by reloading the Consul process, `consul reload` or `kill -HUP <process_id>`.
```

Comment:
```suggestion
A rolling update can be made by restarting the Consul agents (clients and servers) in turn. `consul reload` or `kill -HUP <process_id>` is _not_ sufficient to change the gossip configuration.
```

@kaitlincarter-hc I'm pretty sure gossip encryption needs a full restart from my recent poking there, if you tried it and know differently please let me know.